Solicitors can see all Defence Requests for their Firm

robmckinnon opened the pull request:
* Implements: https://www.pivotaltracker.com/story/show/91086290
* Add `related_to_solicitor()` method to DefenceRequest that returns solicitor's firms defence requests.
* Assumes that solicitor's `organisation_uid` is always present and correct on defence requests that have `solicitor_uid == solicitor's uid`. This needs to be verified when we implement assigning a solicitor to a request.
* Having to write `solicitor.organisation_uids.first` to get solicitor `organisation_uid` - suggestions welcome for a better interface.
* Add `login_with()` helper to allow custom user fields to be mocked on profile.

pedrosmmoreira commented on app/models/defence_request.rb:
```diff
@@ -10,8 +10,14 @@ class DefenceRequest < ActiveRecord::Base
 
-  def self.has_solicitor(solicitor)
-    where(solicitor_uid: solicitor.uid)
+  class << self
```
What is the motivation for switching class method syntax here? This would, for instance, raise issues with Rubocop or other style checkers, adds noise to the class and seems a bit spurious?

robmckinnon replied:
Think deleting has_solicitor() is a good idea. I'll check first that it is not used elsewhere.

pedrosmmoreira commented on spec/features/solicitor/dashboard_spec.rb:
```diff
@@ -8,14 +8,16 @@
+    not_accepted_defence_request = create :defence_request, offences: "Extortion"
 
-    login_with_role "solicitor", solicitor_user.uid
+    login_with_user solicitor_user
```
Any reason for changing this method name? I'm fine with passing less parameters around, but if feels the original name provided more context.

robmckinnon replied:
Method takes a user object as a parameter, so `login_with_user` seemed an appropriate name. The `login_with_role` helper method has not been deleted yet.

pedrosmmoreira replied:
Ah, I see. How do you feel about refactoring `login_with_role` to accept the user object and keeping only one method? `login_with_user` feels slightly redundant while letting the test reader know that there are several types of user exposes important information, if that makes sense?

pedrosmmoreira commented on spec/models/defence_request_spec.rb:
```diff
@@ -165,4 +165,27 @@
     end
   end
+
+  describe "retrieve requests associated with solicitor's firm" do
```
It is preferable to maintain the 4 phase setup in tests as much as possible. Having scenarios that consist entirely of expectations obfuscates reading. In this particular case, there is not much to gain from extracting the before block, for instance.

robmckinnon replied:
Have removed before block as suggested.

pedrosmmoreira commented:
Some style related comments, otherwise looks good to me.

robmckinnon commented:
As agreed with @pedrosmmoreira and @ansonK :
* Renamed `login_with_user()` to `login_with()`
* Replaced `login_with_remote()` usage with `login_with()`

Should be ready for merge now.
